fip-6.md

```diff
@@ -0,0 +1,376 @@
+* traverse list until remaining locked amount > 0 or 50 entries removed.
+* remove entry with remaining locked amound == 0
+* return status ok and removed_count
+
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|No work|No items to remove.|403|||"No work"|
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+||status|String|Ok|
+||removed_count|String|number of items removed|
+###### Example
+```
+{
+   "status": "Ok",
+   "removed_count": "25"        
+}
+```
+
+## Specification
+### Clean locks
+#### New end point: *clean_locks* 
+#### New action in new fio.lock contract cleanlocks
+, 
+##### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|actor|Yes|FIO account name|FIO account for the signer|
+
+
+##### Processing
+* require auth actor.
+* read the locktokens table by the specified account
+* traverse all locks..
+* if entry has remaining locked amount > 0 leave this lock.
+* remove entry with remaining locked amount == 0
+* throw no work error if no locks found to remove.
+* allow this to be called once per day.
+* return status ok and removed_count
+
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|No work|No items to remove.|400|||"No work"|
+|time violation|time between calls exceeded.|400|||"Time violation"|
+
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+||status|String|Ok|
+||removed_count|String|number of items removed|
+###### Example
+```
+{
+"status": "Ok",
+"removed_count": "25"        
+}
+```
+
+## Specification
+### get locks
+#### New end point: *get_locks* 
+#### New action in new fio.lock contract getlocks
+, 
+##### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_public_key|Yes|fio public key, see FIO public key validation rules.|FIO public key to search locks.|
+|offset|no|number greater or equal 0.|offset at which to begin results|
+|limit|no|number greater than 0|number of results to display|
+
+
+##### Processing
+* hash the pub key.
+* get the account from the fio account mapping
+* ensure hash and account match.
+* read the locktokens table by the specified account
+* traverse all locks..
+* add each lock to the results.
+* provide paging for results.
+* return list of locks.
+
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|No results found|No items found.|403|||"No results"|
+
+
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+||status|String|Ok|
+||locks|String|json formatted list of locks|
+###### Example
+```
+{
+  "locks": 
+          [
+           {
+            owner: edrgrdftrgyt
+            lock_amount: 1000,000000000
+            payouts_performed: 2
```

Comment:
What is this field telling me? You need to define it.

Reply:
in the implementation section.